docs/DeveloperGuide.md

```diff
@@ -1,34 +1,610 @@
+|count|Integer indicating how many times the event is to be repeated|
+
+For example, to repeat a personal event located at index 2 for 3 days, the user will key in this: `repeat personal 2 daily 3`.
+
+The following sequence diagram shows the overall process of repeating an event:
+
+![Sequence Diagram for Repeat Command](./diagrams/RepeatScenario.jpg)
+
+The steps to repeating the command will be described in the following sections. 
+
+##### Step 1: Obtaining the event information
+
+The repeat command first locates the event that is to be repeated along with its starting date.
+Next, it creates an empty ArrayList called `repeatEventList` that will contain all repeated events.
+
+![Sequence Diagram for Repeat Command step 1](./diagrams/repeatstep1.jpg)
+
+##### Step 2: Incrementing the Original Date
+
+The repeatCommand object will now call upon Event to increment the date by a fixed unit of time. In the following diagram, it shows how the program flow will work should a monthly repeat increment be set. 
+The increment date will be assigned the name `repeatDate`
+
+![Sequence Diagram for Repeat Command step 2](./diagrams/repeatstep2.jpg)
+
+##### Step 3: Set Repeat type
+
+The repeatCommand now will set the original Event to have a repeat type setting. In the diagram's example, it will be set to `monthly`
+
+![Sequence Diagram for Repeat Command step 3](./diagrams/repeatstep3.jpg)
+
+##### Step 4: Clone the event
+
+The original event is now cloned as shown in the following sequence diagram.
+
+![Sequence Diagram for Repeat Command step 4](./diagrams/repeatstep4.jpg)
+
+##### Step 5: Add the cloned activity
+
+The cloned activity will now have its date set to be `repeatDate`. This new activity is now added into the `repeatEventList` as shwon in the following diagram.
+
+![Sequence Diagram for Repeat Command step 5](./diagrams/repeatstep5.jpg)
+
+##### Step 6: Repeat steps 2 to 5
+
+Steps 2 to 5 are repeated up till the number specified by `count` as shown in the loop.
+
+![Sequence Diagram for Repeat Command step 6](./diagrams/repeatstep6.jpg)
+
+##### Step 7: Assign the ArrayList as the RepeatEventList attribute of the original event.
+
+Finally, set the `repeatEventList` using the `setRepeatEventList` command as shown in the following section of the sequence diagram. The results of this process is printed and control returns back to the main program.
+
+![Sequence Diagram for Repeat Command step 7](./diagrams/repeatstep7.jpg)
+
+#### Deadline feature
+
+The user executes ```deadline 1; 7/10/20; 11:20 PM``` command to set the deadline for the 1st event in Personal event list
+to be on the 7th October 2020 at 11:20 PM. 
+The ```deadline``` command calls ```DeadlineCommand#execute()```, adding/updating the personal event deadline. <br>
+Given below is how the deadline command behave: <br>
+
+<p align="center">
+  <img width="414" height="562" src="./diagrams/DeadlineScenario.jpg">
```

Comment:
![image](https://user-images.githubusercontent.com/60382285/97660415-7126d700-1aac-11eb-8b23-9df99d628622.png)
should the states in the state diagrams be full boxes instead of the half boxes here?